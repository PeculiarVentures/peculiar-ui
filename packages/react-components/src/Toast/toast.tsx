import * as React from 'react';
import { Alert, AlertProps } from '../Alert';
import { Fade } from '../Fade';

/**
 * Types.
 */
interface ToastOwnProps {
  /**
   * The `id` of the toast. Mostly used when you need to prevent duplicate.
   */
  id: string;
  /**
   * The content of the component.
   */
  children: React.ReactNode;
  /**
   * Callback fired when the component requests to be closed.
   */
  onClose: (id: string) => void;
  /**
   * The delay before the toast hides (in milliseconds). If set to `null`, toast will never dismiss.
   */
  duration?: number | null;
  /**
   * If `true`, toast will show a close button.
   */
  isClosable?: boolean;
  /**
   * Props applied to the `Alert` element.
   */
  alertProps?: Omit<AlertProps, 'children' | 'onClose'>;
};
/**
 *
 */

export const Toast: React.FC<ToastOwnProps> = (props) => {
  const {
    id,
    duration,
    isClosable,
    alertProps,
    children,
    onClose,
  } = props;

  const handleClose = () => {
    onClose(id);
  };

  React.useEffect(() => {
    if (duration > 0) {
      const timeout = window.setTimeout(
        handleClose,
        duration,
      );

      return () => {
        clearTimeout(timeout);
      };
    }

    return undefined;
  }, []);

  return (
    <Fade in>
      <Alert
        {...alertProps}
        onClose={isClosable ? handleClose : undefined}
      >
        {children}
      </Alert>
    </Fade>
  );
};

Toast.displayName = 'Toast';

Toast.defaultProps = { duration: 4000 };
