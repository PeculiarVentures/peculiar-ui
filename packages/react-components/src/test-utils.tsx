import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
