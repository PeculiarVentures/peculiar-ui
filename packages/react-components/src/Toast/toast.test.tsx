import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Toast } from './toast';

describe('<Toast />', () => {
  it('should render with default styles', () => {
    const onClose = jest.fn();

    const { asFragment } = render(
      <Toast
        id="toast1"
        onClose={onClose}
      >
        Inside
      </Toast>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
