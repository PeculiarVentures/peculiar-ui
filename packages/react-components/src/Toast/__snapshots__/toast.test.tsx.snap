// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: var(--pv-size-base-2) var(--pv-size-base-4);
  box-sizing: border-box;
  border-radius: 4px;
  background-color: var(--pv-color-black);
}

.emotion-1 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  padding: var(--pv-size-base) 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-height: var(--pv-text-b3-height);
}

<div
    class="emotion-0"
    role="alert"
    style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <p
      class="emotion-1"
    >
      Inside
    </p>
  </div>
</DocumentFragment>
`;
